### PR TITLE
add python3.9 support

### PR DESCRIPTION
**Pull request by freddrake**

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -5,8 +5,8 @@
 3.6.0 (unreleased)
 ==================
 
-- Added support for Python 3.8.  This primarily involves avoiding the
-  new-in-3.8 validation of the format string when using the
+- Added support for Python 3.8, 3.9.  This primarily involves avoiding
```
Comment (stevepiercy) [inline CHANGES.rst]:
```suggestion
- Added support for Python 3.8 and 3.9.  This primarily involves avoiding
```

**Comment by jugmac00** [comment]
I think support for Python 3.4 could also be dropped.

@icemac / @jamadden / @mgedmin  What is the current philosophy / which versions should be supported?

**Comment by stevepiercy** [comment]
For reference:
- [Python Development Cycle - End-of-life branches](https://devguide.python.org/devcycle/#end-of-life-branches)
- [Python Developer’s Guide - Status of Python branches](https://devguide.python.org/#status-of-python-branches)

Of course Zope might have different guidelines, or none at all.

**Comment by icemac** [comment]
Currently we drop Python 3.4 support everywhere as it is beyond its end of live. Python 2 support is still kept to ease migration.

**Comment by freddrake** [comment]
I've no position on dropping / retaining Python 3.4 support so long as the tests still pass (and yes, I checked).  That can be done as a separate PR if anyone is sufficiently interested.

**Pull request merged**